src/Bicep.Core.IntegrationTests/NestedRuntimeMemberAccessTests.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.Core.IntegrationTests
+{
+    [TestClass]
+    public class NestedRuntimeMemberAccessTests
+    {
+        [TestMethod]
+        public void Nested_runtime_member_access_is_not_supported()
+        {
+            var result = CompilationHelper.Compile(@"
+resource appPlan 'Microsoft.Web/serverfarms@2020-12-01' existing = {
+  name: 'foo'
+}
+
+resource virtualMachine 'Microsoft.Compute/virtualMachines@2020-12-01' = {
+  name: 'name'
+  location: 'westus'
+
+  resource windowsVMDsc 'extensions' existing = {
+    name: appPlan.properties.provisioningState
+  }
+}
+
+resource storage 'Microsoft.Storage/storageAccounts@2021-02-01' existing = {
+  name: appPlan.properties.provisioningState
+}
+
+var foo = virtualMachine::windowsVMDsc.properties.autoUpgradeMinorVersion
+var bar = storage.properties.accessTier
+var baz = storage.listKeys().keys
+");
+            result.ExcludingLinterDiagnostics().Should().HaveDiagnostics(new[]
+            {
+                ("BCP307", DiagnosticLevel.Error, @"The expression may lead to nested runtime functions, which is not supported. Accessible properties of windowsVMDsc include ""apiVersion"", ""id"", ""name"", ""type""."),
+                ("BCP307", DiagnosticLevel.Error, @"The expression may lead to nested runtime functions, which is not supported. Accessible properties of storage include ""apiVersion"", ""id"", ""name"", ""type""."),
+                ("BCP307", DiagnosticLevel.Error, @"The expression may lead to nested runtime functions, which is not supported. Accessible properties of storage include ""apiVersion"", ""id"", ""name"", ""type""."),
```

Comment:
>""apiVersion"", ""id"", ""name"", ""type""." [](http://example.com/codeflow?start=170&length=44)

I'm confused by this error message. Normally the accessible properties would be the full schema of the resource. It's only because the name is using a non-DTC is why we only let you access apiVersion, id, name, and type, right?

Reply:
Correct. If the name contains a non-DTC value, referencing `$.properties.*` will produce nested reference functions.

Reply:
I don't know if it's obvious what nested runtime functions means and why if the user doesn't understand how the JSON works, but I'm also not sure what the solution is. 

One option could be something like "The expression leads to unsupported nested runtime functions because resource <symbolic name> has <property> set to a runtime expression. In this situation, the accessible properties of...." However, <property> could be `name` or could be `scope`, or both.

Reply:
Yeah...it's hard to explain what the error is if the user doesn't understand the semantic of the underlying JSON template. I like the suggested error message, it's slightly more verbose, but it's less confusing. We don't have to worry about `name` vs `scope`, because existing resources still require the `scope` property to be a DTC

I'm wondering if we should hide the concept "nested functions` from users if we cannot do a good job explaining it in a short error message. What do you think the one below?

> The expression cannot be evaluated, because the "name" property of the referenced existing resource contains a value that cannot be determined at the start of the deployment. In this situation, the accessible properties of "xyz" include....

Reply:
Nice. That works! 

Separately, it'd be good to enclose the symbolic name in the "accessible properties" part in double quotes as well.

Reply:
Updated the error message.